### PR TITLE
Add news items to sanity, update ui

### PR DESCRIPTION
**Pull request by kcbreuil**

### CODE REVIEW

File: web/src/components/newsSection.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                        />
+                    );
+                })}
+            </div>
+        </section>
+    );
+}
+
+NewsSection.propTypes = {
+    pathname: PropTypes.string.isRequired,
```
Comment (asasmith) [inline web/src/components/newsSection.js:25]: Remove

File: web/src/components/newsList.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import MoreButton from './moreButton';
 
 const Carousel = loadable(() => import('./carouselWrapper'));
 
-export default function NewsList({ news }) {
+export default function NewsList({ news, pathname }) {
```
Comment (asasmith) [inline web/src/components/newsList.js:15]: Remove `pathname`

File: web/src/components/newsList.js [hunk 3]
```diff
@@ -55,7 +55,11 @@ export default function NewsList({ news }) {
                 >
                     {news.map((newsItem) => {
                         return (
-                            <NewsCard key={newsItem.title} data={newsItem} />
+                            <NewsCard
+                                key={newsItem.title}
+                                data={newsItem}
+                                pathname={pathname}
```
Comment (asasmith) [inline web/src/components/newsList.js:61]: Remove

File: web/src/components/newsList.js [hunk 4]
```diff
@@ -66,4 +70,5 @@ export default function NewsList({ news }) {
 
 NewsList.propTypes = {
     news: PropTypes.array.isRequired,
+    pathname: PropTypes.string.isRequired,
```
Comment (asasmith) [inline web/src/components/newsList.js:73]: Remove

File: web/src/components/newsContent.js [hunk 5]
```diff
@@ -1,33 +1,24 @@
...
-                posuere. Maecenas in lorem commodo, varius neque vitae, vehicula
-                dolor. Sed ut viverra eros. Orci varius natoque penatibus et
-                magnis dis parturient montes, nascetur ridiculus mus. Nam
-                convallis arcu a quam tristique placerat. Maecenas egestas, nibh
-                et ultricies tristique, purus magna sodales nisi, sit amet
-                efficitur justo massa accumsan augue. Proin condimentum dolor
-                elementum augue fermentum pretium. Phasellus consequat diam vel
-                ante placerat fermentum sit amet vitae velit. Quisque vel ante
-                ac erat accumsan tincidunt.
+            <p className="leading-relaxed prose">
```
Comment (asasmith) [inline web/src/components/newsContent.js:15]: I think this should be a `div`. The `<BlockText>` component renders `<div>`'s and they can't be descendants of `<p>`'s

File: web/src/components/newsCard.js [hunk 6]
```diff
@@ -3,18 +3,21 @@ import PropTypes from 'prop-types';
 import { GatsbyImage, getImage } from 'gatsby-plugin-image';
 import { Link } from 'gatsby';
 
-export default function NewsCard({ data }) {
+export default function NewsCard({ data, pathname }) {
```
Comment (asasmith) [inline web/src/components/newsCard.js:6]: Remove `pathname`

File: web/src/components/newsCard.js [hunk 7]
```diff
@@ -3,18 +3,21 @@ import PropTypes from 'prop-types';
...
+        },
     } = data;
 
-    const imageData = getImage(asset);
+    const imageData = getImage(heroImage.asset);
 
     return (
         <Link
-            to="/"
+            to={`${pathname}/${current}`}
```
Comment (asasmith) [inline web/src/components/newsCard.js:20]: Should be
```
to={`/news/${current}`}
```

File: web/src/components/newsCard.js [hunk 8]
```diff
@@ -32,4 +35,5 @@ export default function NewsCard({ data }) {
 
 NewsCard.propTypes = {
     data: PropTypes.object.isRequired,
+    pathname: PropTypes.string.isRequired,
```
Comment (asasmith) [inline web/src/components/newsCard.js:38]: Remove

**Review by asasmith — CHANGES_REQUESTED** [review]

File: web/src/templates/newsTemplate.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    query NewsTemplateQuery($id: String!) {
+        newsItem: sanityNewsItem(id: { eq: $id }) {
+            heroImage {
+                asset {
+                    gatsbyImageData
+                }
+            }
+            id
+            newsTitle
+            newsSubTitle
```
[the remark this answers is not in the document]
Comment (kcbreuil) [reply web/src/templates/newsTemplate.js:37]: @asasmith Yes for the article page. newsContent component

File: web/src/components/newsList.js [hunk 10]
```diff
@@ -55,7 +55,10 @@ export default function NewsList({ news }) {
                 >
                     {news.map((newsItem) => {
                         return (
-                            <NewsCard key={newsItem.title} data={newsItem} />
+                            <NewsCard
+                                key={newsItem.title}
+                                data={newsItem.node}
```
Comment (asasmith) [inline web/src/components/newsList.js:60]: Destructure `title` and `node`

File: web/src/components/newsSection.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import PropTypes from 'prop-types';
+
+import NewsCard from './newsCard';
+
+export default function NewsSection({ news }) {
+    return (
+        <section className="container my-14">
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 2xl:grid-cols-4 place-items-center gap-9">
+                {news.map((item) => {
+                    return <NewsCard data={item.node} key={item.node.id} />;
```
Comment (asasmith) [inline web/src/components/newsSection.js:11]: Destructure `node`

File: web/src/templates/newsTemplate.js [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        news: { edges },
+    } = data;
+
+    const { heroImage, newsTitle } = newsItem;
+
+    const { pathname } = location;
+
+    const posts = edges.map((item) => item.node);
+
+    const filteredPosts = posts.filter((item) => item.id !== newsItem.id);
```
Comment (asasmith) [inline web/src/templates/newsTemplate.js:22]: Can this be done in 1 line?
```
const filteredPosts = edges.filter((post) => post.node.id !== newsItem.id)
```

File: web/src/templates/newsTemplate.js [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+export default function NewsTemplate({ data, location }) {
+    const {
+        newsItem,
+        news: { edges },
+    } = data;
+
+    const { heroImage, newsTitle } = newsItem;
+
+    const { pathname } = location;
```
Comment (asasmith) [inline web/src/templates/newsTemplate.js:18]: Do we still need pathname? I don't see where it's being used in [recentNewsList.js](https://github.com/bckrm/airtec/blob/main/web/src/components/recentNewsList.js)

File: web/src/templates/newsTemplate.js [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+    const posts = edges.map((item) => item.node);
+
+    const filteredPosts = posts.filter((item) => item.id !== newsItem.id);
+
+    return (
+        <Layout>
+            <Hero image={heroImage} title={newsTitle} />
+            <NewsContent data={newsItem} />
+            <RecentNewsList news={filteredPosts} pathname={pathname} />
```
Comment (asasmith) [inline web/src/templates/newsTemplate.js:28]: Same question about `pathname` as above

**Review by asasmith — CHANGES_REQUESTED** [review]

File: web/src/templates/newsTemplate.js [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    const {
+        newsItem,
+        news: { edges },
+    } = data;
+
+    const { heroImage, newsTitle } = newsItem;
+
+    const posts = edges
+        .map((post) => post.node)
+        .filter((post) => post.id !== newsItem.id);
```
Comment (asasmith) [inline web/src/templates/newsTemplate.js:20]: `filter` creates a new array so I don’t think you need `map`. Stick with the `filteredPosts` naming convention so it’s clear what this array is. 

Also, destructure `id` from `newsItem`

**Review by asasmith — CHANGES_REQUESTED** [review]

File: web/src/templates/newsTemplate.js (hunk 15 above)
Comment (kcbreuil) [reply web/src/templates/newsTemplate.js:20]: @asasmith that didn't work for me when I tried over the weekend. it was not pulling anything out from the array

Comment (asasmith) [reply web/src/templates/newsTemplate.js:20]: `.filter((post) => post.node.id !== newsItem.id` should work. 

You'll also have to adjust `recentNewsList.js` to destructure `node` in the `map` method (similar to `newsList.js` [here](https://github.com/bckrm/airtec/blob/f468dfab9cbcc6ee393f2aa8f384a711d7e33356/web/src/components/newsList.js#L56-L59))

**Review by asasmith — APPROVED** [review]

**Pull request merged**